### PR TITLE
Implement identical logic using Vite

### PR DESCRIPTION
**Pull request by fwouts**
This removes the VFS fonts because they fail to build with Vite (and likely also with Rollup, which Vite uses under the hood).

See https://github.com/bpampuch/pdfmake/issues/2654 for some potential ideas to resolve this.

### CODE REVIEW

File: package/src/dt_with_buttons.js [hunk 1]
```diff
@@ -1,14 +1,11 @@
-import jQuery from 'jquery';
-import JSZip from 'jszip';
-import pdfMake from 'pdfmake';
-import pdfFonts from 'pdfmake/build/vfs_fonts';
-import DataTable from 'datatables.net-bs5';
-import 'datatables.net-buttons-bs5';
-import 'datatables.net-buttons/js/buttons.html5';
- 
+import DataTable from "datatables.net-bs5";
```
Comment (fwouts) [inline package/src/dt_with_buttons.js:1]: Note: the reordering is just because my IDE automatically sorts imports. Feel free to revert!

Comment (mwouts) [reply package/src/dt_with_buttons.js:1]: No pb at all!

**Comment by mwouts** [comment]
This is great, thank you @fwouts ! I could not decide which of `rollup`, `esbuild` or `vite` to use, so I am glad you made a choice! No problem for the pdf fonts, really not a problem here - my focus here is just to learn how to bundle the library.

One more question. I see that `vite-plugin-singlefile` produces a single HTML file, that's almost what I need, but do you think instead we could bundle a single JS file from which I could import e.g. `import {DataTable, jQuery as $} from "bundle.js"`? I want to keep that import separate from `data` and `dt_args` that are provided at a later stage.

**Comment by fwouts** [comment]
Yes! All you need is to remove the single file plugin, and it will do exactly that 😁

**Comment by mwouts** [comment]
> Yes! All you need is to remove the single file plugin, and it will do exactly that 😁

Almost there! Still two points that I would like to address when time permits:
1. The `data` that is defined in the HTML file currently goes to the bundled `.js` file. I would need to keep it appart and bundle just the `./src/dt_with_buttons.js` file. I want to be able to replace `import {DataTable, jQuery as $} from "./src/dt_with_buttons.js";` (which has external dependencies) in the original HTML with `import {DataTable, jQuery as $} from "bundle.js";` (self contained, so I can actually import from its base64 representation).
2. The bundle file is called e.g. `index-0M51yzQc.js`. I would need to know or even set the bundle file name in advance.

**Comment by mwouts** [comment]
I have had some success with `esbuild` in (#2) - my only concern is that I could not find how to import from the bundle, so I am currently exporting through `window` - not sure if there is a better way to do so?

**Comment by mwouts** [comment]
Thanks @fwouts for fixing the `esbuild` PR! I've taken that other PR (#2) instead as it is closer to what I will be doing in `itables`.

**Pull request closed without merging**